main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2040,6 +2040,13 @@ object Defaults extends BuildCommon {
     Def.inputTask {
       val handle = bgRunMain.evaluated
       val service = bgJobService.value
+      streams.value.log.info(
+        s"""
+           |Executing runMain in the background.
+           |To terminate the execution, run bgStop ${handle.id}.
+           |To execute runMain in the foreground, use fgRunMain instead.
```

Comment:
Another option is just to assign `run\runMain` as `fgRun\fgRunMain` (which can be terminated via `ctrl+c`). This might be a better solution (adjusting behaviour of `run\runMain` to match what sbt users expect instead of using potentially spammy log to inform them about current behaviour of `run\runMain`).

Looking to hear opinions.

Reply:
Big +1 for just using `fgRun` semantics. I've been using Sbt for almost as long as it has existed (going on a decade and a half?) and this is the very first time I have ever 1) heard about `fgRun`, or 2) learned that `run` actually backgrounds the task and *intentionally* suppresses <kbd>Ctrl</kbd>-<kbd>C</kbd>. Genuinely, I can't think of a single sbt user who would expect the present semantics.

Reply:
I shall leave this PR on for some more time to gather user feedbacks. If the feedbacks are consistent, I will close this PR and open another PR that let `run/runMain` invoke `fgRun/fgRunMain` by default.

Reply:
First off, sorry about the inconvenience this is causing.

Probably the central topic is the idea of non-blocking run (https://github.com/sbt/sbt/issues/6178), which I and others have been wanting for various reasons. The job of sbt is take orders and dispatch tasks to baristas. From that perspective, `run` and `tests` are like cashier going on a long lunch break, and while the person is out, no orders can come in.

That has been ok behavior, but as we're trying to make sbt-as-Metals-backend happen, these long running tasks seems sort of silly because fundamentally `run` and `test` are just running some function in a sandbox (or shell out to new JVM). Neither involves sbt much. So the idea was to:
1. implement `bgRun` (background run) and/or persistent worker
2. encode `run` as `bgRun` <- we are here

The oversight was the cancellation semantics. In case of `run` we should forward cancellation to cancelling the bgRun job.

Reply:
As long as cancelation is forwarded, I'm content!